specs/transport/car/carv2/index.md

```diff
@@ -0,0 +1,177 @@
+   * `4` has the same meaning as `3` except that the IPLD selector is encoded as a DAG-CBOR IPLD block immediately following the Header and is prefixed with a varint indicating the length of the bytes to decode.
+2. **Bit 8** indicating whether the blocks have been properly deduplicated (`1`) and that a reader should not expect any repeated CIDs, or that the blocks _may_ contain duplicates (`0`) (perhaps as a trade-off for speed or memory consumption during generation).
+
+Future amendments of this specification may introduce additional characteristics, including:
+
+* **TODO:** what else are we potentially filling 128-bits up with?
+
+### CARv1 data payload
+
+The CARv1 data payload starts at the "Data offset" indicated by the CARv2 header. It is important that a decoder adhere to this offset rather than assuming the data payload begins immediately after the header in order to make allowance for additional data following the header as indicated by the characteristics field in future amendments to this specification.
+
+The CARv1 data payload ends after the number of bytes indicated by the "Data length" field in the header. This is important as the CARv1 parser has no means to determine the end of the container other than encountering the end of a byte stream. Therefore, a CARv2 decoder may defer to a CARv1 decoder to load the `CID:Bytes` sequence payload and "roots" array, but must be able to deliver it a byte stream that strictly begins at "Data offset" and ends at "Data length".
+
+The CARv1 data payload must include a valid CARv1 header, including a roots array and a `version` field with a value strictly of `1`.
+
+Refer to the [CARv1 Specification](../carv1/index.md) for details on the CARv1 format.
+
+### Index payload
+
+The CARv2 index payload follows the CARv1 data payload, but may be offset by padding or additional data as dictated by the characteristics field. The index payload begins at "Index offset", which _must_ be after the end of the CARv2 data payload, and continues until the end of the CARv2 byte stream (its length is not encoded in the header).
+
+An "Index offset" value of `0` in the CARv2 header indicates that there is no index in this CARv2 container and no attempt should be made to read it.
+
+### Index format
+
+The CARv2 index is a flexible container format itself, allowing for different index layouts depending on suitability for a particular application or set of data - generation speed, usage performance, size, etc. Supported index formats will be detailed in this specification, below.
+
+Index data, once read from the CARv2 container, provides a mapping of hash digest bytes to block location in byte offset from the beginning of the CARv2 container. The index only uses the hash digest&mdash;it does not use the full bytes of a CID, nor does it use any of the multihash prefix bytes.
```

Comment:
did you mean to write an HTML-escaped dash here? I think a `-`, `--`, or `;` would work in plaintext, but maybe I'm missing something.

Reply:
Yes, I did, cause I was doing it on Linux where I can never remember the keyboard shortcut for em-dash, so this is how I insert them in that situation. (I'm a stickler for proper em-dashes and their usage). Have fixed now.